### PR TITLE
feat(ui): extend external ids for projects and components

### PR DESCRIPTION
**Pull request by maierthomas**
- add external ids to components (view/edit/merge)
- add external id in the edit page of the project

### CODE REVIEW

File: frontend/sw360-portlet/src/main/webapp/html/components/edit.jsp [hunk 1]
```diff
@@ -155,6 +157,20 @@ require(['jquery', 'modules/sw360Validate', 'modules/autocomplete', 'modules/con
...
             function () {
+                var valid = true;
+                $.each($('#externalIdsTable tr.bodyRow'), function (i, row) {
+                    var key = $('#'+row.id+' td input.keyClass').val();
+                    var value = $('#'+row.id+' td input.valueClass').val();
+                    var isOneEmpty = ((key !== '') && (value === '')) || ((key === '') && (value !== ''));
+                    if (isOneEmpty) {
+                        $.alert('There is a blank field in one or more of your external ids. Please correct this before updating the component.');
+                        valid = false;
+                        return false;
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/html/components/edit.jsp]: returned value is ignored. Instead an environmental variable is modified. That looks not optimal.

File: frontend/sw360-portlet/src/main/webapp/html/projects/edit.jsp [hunk 2]
```diff
@@ -168,6 +170,20 @@ require(['jquery', 'modules/sw360Validate', 'modules/confirm' ], function($, sw3
...
             function () {
+                var valid = true;
+                $.each($('#externalIdsTable tr.bodyRow'), function (i, row) {
+                    var key = $('#'+row.id+' td input.keyClass').val();
+                    var value = $('#'+row.id+' td input.valueClass').val();
+                    var isOneEmpty = ((key !== '') && (value === '')) || ((key === '') && (value !== ''));
+                    if (isOneEmpty) {
+                        $.alert('There is a blank field in one or more of your external ids. Please correct this before updating the project.');
+                        valid = false;
+                        return false;
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/html/projects/edit.jsp]: duplicated code!

File: frontend/sw360-portlet/src/main/webapp/html/components/edit.jsp [hunk 3]
```diff
@@ -155,6 +157,20 @@ require(['jquery', 'modules/sw360Validate', 'modules/autocomplete', 'modules/con
 
         $('#formSubmit').click(
             function () {
+                var valid = true;
+                $.each($('#externalIdsTable tr.bodyRow'), function (i, row) {
```
Comment (maxhbr) [inline frontend/sw360-portlet/src/main/webapp/html/components/edit.jsp]: @alexbrdn : replace by vaidate->notEmpty

File: frontend/sw360-portlet/src/main/webapp/html/components/edit.jsp (hunk 1 above)
Comment (maierthomas) [reply frontend/sw360-portlet/src/main/webapp/html/components/edit.jsp]: js removed and added required/minlength to the external id input fields

File: frontend/sw360-portlet/src/main/webapp/html/projects/edit.jsp (hunk 2 above)
Comment (maierthomas) [reply frontend/sw360-portlet/src/main/webapp/html/projects/edit.jsp]: js removed and added required/minlength to the external id input fields

**Pull request merged**